boards/shared/Inc/Io/Io_SharedFrequencyOnlyPwmInput.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <stm32f3xx_hal.h>
+#include <stdbool.h>
+#include <assert.h>
+
+#include "Io_SharedFrequencyOnlyPwmInputConfig.h"
+
+#define MAX_NUM_OF_FREQ_ONLY_PWM_INPUT 2
+
+#ifndef MAX_NUM_OF_FREQ_ONLY_PWM_INPUT
+#error Missing definition: MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS must be defined in Io_SharedFreqOnlyPwmInputConfig.h.
+#endif
+
+struct FrequencyOnlyPwmInput;
+
+/**
+ * Allocate and initialize a PWM input using the given (hardware) timer
+ *
+ * @note The given timer must be initialized with:
+ *       - Input Capture direct mode on Channel A
+ *       - Input Capture direct mode on Channel B
+ *
+ * @param htim: The handle of the timer measuring the PWM input
+ * @param timer_frequency_hz: The frequency of the timer measuring the PWM input
+ * @param rising_edge_tim_channel: The rising edge channel of the timer
+ * measuring the PWM input
+ * @return Pointer to the allocated and initialized PWM input
+ */
+struct FrequencyOnlyPwmInput *Io_SharedFrequencyOnlyPwmInput_Create(
+    TIM_HandleTypeDef *htim,
+    float              timer_frequency_hz,
+    uint32_t           rising_edge_tim_channel);
+
+/**
+ * Get the frequency for the
+￼* @param previous_rising_edge: The input capture value of the previous
+risingiven PWM input
+ * @param frequency_only_pwm_input: The PWM input to get frequency for
+ * @return The frequency for the given PWM input
+ */
+float Io_SharedFrequencyOnlyPwmInput_GetFrequency(
```

Comment:
So this was my bad.

Adding `const` after `*` is useless in the declaration. It only works in the definition. 

i.e. 
```
void foo(int* const x); // delcaration with useless const
void f00(int * const x)  // this one actually does something
{ // do something }

Reply:
This is fixed